chainstate/src/detail/transaction_verifier/mod.rs

```diff
@@ -643,6 +632,29 @@ impl<'a, S: BlockchainStorageRead> TransactionVerifier<'a, S> {
         inputs.iter().try_for_each(|input| self.fetch_and_cache(input.outpoint()))
     }
 
+    fn block_reward_outputs(
+        &self,
+        block_id: &Id<GenBlock>,
+    ) -> Result<Vec<TxOutput>, ConnectTransactionError> {
+        match block_id.classify(self.chain_config) {
+            GenBlockId::Genesis(_) => Ok(self
+                .chain_config
+                .genesis_block()
+                .block_reward_transactable()
+                .outputs()
+                .unwrap_or(&[])
+                .to_vec()),
+            GenBlockId::Block(id) => Ok(self
+                .db_tx
+                .get_block(id)?
```

Comment:
Getting the whole block just for the reward... not the best. Please throw in a TODO there to investigate how to improve this later.

Reply:
https://github.com/mintlayer/mintlayer-core/issues/344